src/main/java/org/alliancegenome/curation_api/services/associations/constructAssociations/ConstructGenomicEntityAssociationService.java

```diff
@@ -0,0 +1,179 @@
+		associationIdStrings.removeIf(Objects::isNull);
+		List<Long> associationIds = associationIdStrings.stream().map(Long::parseLong).collect(Collectors.toList());
+		
+		return associationIds;
+	}
+
+	@Transactional
+	public ConstructGenomicEntityAssociation deprecateOrDeleteAssociation(Long id, Boolean throwApiError, String loadDescription, Boolean deprecate) {
+		ConstructGenomicEntityAssociation association = constructGenomicEntityAssociationDAO.find(id);
+		
+		if (association == null) {
+			String errorMessage = "Could not find ConstructGenomicEntityAssociation with id: " + id;
+			if (throwApiError) {
+				ObjectResponse<ConstructGenomicEntityAssociation> response = new ObjectResponse<>();
+				response.addErrorMessage("id", errorMessage);
+				throw new ApiErrorException(response);
+			}
+			log.error(errorMessage);
+			return null;
+		}
+		if (deprecate) {
+			if (!association.getObsolete()) {
+				association.setObsolete(true);
+				if (authenticatedPerson != null) {
+					association.setUpdatedBy(personDAO.find(authenticatedPerson.getId()));
+				} else {
+					association.setUpdatedBy(personService.fetchByUniqueIdOrCreate(loadDescription));
+				}
+				association.setDateUpdated(OffsetDateTime.now());
+				return constructGenomicEntityAssociationDAO.persist(association);
+			}
+		}
+		
+		List<Note> notesToDelete = association.getRelatedNotes();
+		if (CollectionUtils.isNotEmpty(notesToDelete))
+			notesToDelete.forEach(note -> constructGenomicEntityAssociationDAO.deleteAttachedNote(note.getId()));
+		constructGenomicEntityAssociationDAO.remove(association.getId());
+		
+		
+		return null;
+	}
+	
+	public ObjectResponse<ConstructGenomicEntityAssociation> getAssociation(Long constructId, String relationName, String genomicEntityCurie) {
+		ConstructGenomicEntityAssociation association = null;
+		
+		Map<String, Object> params = new HashMap<>();
+		params.put("subject.id", constructId);
+		params.put("relation.name", relationName);
+		params.put("object.curie", genomicEntityCurie);
+
+		SearchResponse<ConstructGenomicEntityAssociation> resp = constructGenomicEntityAssociationDAO.findByParams(params);
+		if (resp != null && resp.getSingleResult() != null)
+			association = resp.getSingleResult();
+		
+		ObjectResponse<ConstructGenomicEntityAssociation> response = new ObjectResponse<>();
+		response.setEntity(association);
+		
+		return response;
+	}
+	
+	private void addAssociationToConstruct(ConstructGenomicEntityAssociation association) {
+		Construct construct = association.getSubject();
+		List<ConstructGenomicEntityAssociation> currentAssociations = construct.getConstructGenomicEntityAssociations();
+		if (currentAssociations == null)
+			currentAssociations = new ArrayList<>();
+		List<Long> currentAssociationIds = currentAssociations.stream().map(ConstructGenomicEntityAssociation::getId).collect(Collectors.toList());
+		if (!currentAssociationIds.contains(association.getId()));
```

Comment:
Why not using a Set that only collects unique items?

Reply:
It's retrieving a list of IDs for associations attached to a single Construct object, so they will be unique by default.